[WIP] Discover and remember existing icon paths

Code7R opened the pull request:
This avoids blind attempts to access icons in unknown folders.

This also implements a selection option (or excluding option) for
specific (maybe pattern matched) XDG icon themes

Also fixes the detection of wordexp in cmake setup.

Also reorder icon members to save memory (due to type alignment)

Code7R committed:
Discover and remember existing icon paths

This avoids blind attempts to access icons in unknown folders.

This also implements a selection option (or excluding option) for
specific (maybe pattern matched) XDG icon themes

Also fixes the detection of wordexp in cmake setup.

Also reorder icon members to save memory (due to type alignment)

Code7R commented:
@gijsbers Opinions? Works OK for me. I cannot find the original ticket number, though. Also the usage of vector can be changed to YVec or some YArray. unordered_map is tbd, saves some code but could be replaced with some linear searches.

Code7R commented on src/mstring.cc:
```diff
@@ -205,6 +205,24 @@ bool mstring::equals(const mstring &str) const {
     return equals(str.data(), str.length());
 }
 
+bool mstring::requals(const mstring &str) const {
```
Eh, that is dead code. I had a quick idea of using another comparer in the cache, because the difference is easier to spot when you compare backwards. However, current comparer (with memcmp inside) is probably good enough.

Code7R committed:
Actually, get rid of the hashmap

Plain iteration over a few local values should be faster

Code7R committed:
Dropping hardcoded theme subfolders

Also unifying Linux and BSD variants

Code7R committed:
Still using std::vector

It's just more handy

gijsbers commented:
- There is no need to reintroduce HAVE_WORDEXP.
- There is no documentation on IconThemes provided.
- requals is superfluous and can be removed.
- The change to ycollections.h is irrelevant and can be omitted.
- There are "if(".
- OSV("IconThemes".... is not space-aligned with surrounding rows.
- Keep line lengths less than 80 columns.
- Why in yicon.h does findIcon need to go below draw?
- Use a constructor instead of initializing members in the header.

Code7R committed:
Init member from ctor

Code7R committed:
Reformat to match 80char limit and space after control keyword

Also reordered the building of suffix candidates to check better icon
types first, assuming that foo.png would be better than foo_32x32.xpm .

Code7R commented:
* with HAVE_WORDEXP, I guess you meant that it can be auto-assumed? Ok, but I still know a toolchain where it's not available. It's embedded stuff, though, so probably not relevant. Let's drop the guard, ok.
* findIcon() is used from outside, see akeyboard.cc, therefore made public.
* "if(" will be fixed soon. I thought your last complain refered to the multiple ((( not "if (" (because the old code didn't care)
* 80 cols rule... ok, if you say so. But even Linux kernel dropped the rule nowadays.
* will drop ycollection.h changes. It was a cheap knock-off of std::vector anyway, same to requals

Code7R committed:
Drop unused code

Code7R committed:
Assume wordexp availability and less confusing variable naming

Code7R committed:
Fix icon folder lookup in icewm folders

Code7R committed:
Document IconThemes and changed IconPaths settings
